docs/develop/create-sso-office-add-ins-nodejs.md

```diff
@@ -1,14 +1,574 @@
+    - If the POST request does not error, then the response from Azure AD is converted to JSON and sent to the client. This JSON object has an `access_token` property to which Azure AD has assigned the access token to Microsoft Graph.
+
+    ```javascript
+        const stsDomain = 'https://login.microsoftonline.com';
+        const tenant = 'common';
+        const tokenURLSegment = 'oauth2/v2.0/token';
+
+        try {
+            const tokenResponse = await fetch(`${stsDomain}/${tenant}/${tokenURLSegment}`, {
+                method: 'POST',
+                body: form(formParams),
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/x-www-form-urlencoded'
+                }
+            });
+            const json = await tokenResponse.json();
+            
+            res.send(json);
+        }
+        catch(error) {
+            res.status(500).send(error);
+        }
+    }
+    ```
+
+1. Save and close the file.
+
+### Create the route that will fetch the data from Microsoft Graph
+
+1. Open the file `app.js` in the root of the project. Just below the route for '/dialog.html', add the following route. This route is called by the `makeGraphApiCall` function that you created in an earlier step.
+
+    ```javascript
+    app.get('/getuserdata', async function(req, res, next) {
+        
+        // TODO 15: Send a request to the Microsoft Graph REST endpoint.
+
+        // TODO 16: Trim excess information from the returned data and relay it
+        //          to the client.
+        
+    });
+    ```
+
+1. Replace `TODO 15` with the following. About this code, note:
+
+    - The caller of this route, `makeGraphApiCall`, added the access token to Microsoft Graph to the HTTP Request as a header named "access_token".
+    - The `getGraphData` function is defined in the `msgraph-helper.js` file. (This is not the same function as the client-side `getGraphData` function that you defined in the `ssoAuthES6.js` file.)
+    - The last parameter, for `queryParamsSegment`, is hardcoded. If you reuse this code in a production add-in and any part of `queryParamsSegment` comes from user input, be sure that it is sanitized so that it cannot be used in a Response header injection attack.
+    - The code minimizes the data that must come from Microsoft Graph by specifying only the property we need ("name") and only the top 10 folder or file names.
+
+    ```javascript
+    const graphToken = req.get('access_token');    
+    const graphData = await getGraphData(graphToken, "/me/drive/root/children", "?$select=name&$top=10");
+    ```
+
+1. Replace `TODO 16` with the following. About this code, note:
+
+    - If Microsoft Graph returns an error, such as invalid or expired token, there will be a code property in the returned object set to a HTTP status (e.g. 401). The code relays the error to the client. It will br caught in the `.fail` callback of `makeGraphApiCall`.
```

Comment:
```suggestion
    - If Microsoft Graph returns an error, such as invalid or expired token, there will be a code property in the returned object set to a HTTP status (e.g., 401). The code relays the error to the client. It will be caught in the `.fail` callback of `makeGraphApiCall`.
```